imap_processing/codice/codice_l1a.py

```diff
@@ -163,16 +182,18 @@ def get_acquisition_times(self):
         ]
 
         # Get the appropriate values
-        # TODO: update lo_stepping_values.csv with updated data
         lo_stepping_values = lo_stepping_data[
             lo_stepping_data["table_num"] == lo_stepping_table_id
         ]
 
         # Get the acquisition times
-        self.acquisition_times = lo_stepping_values.acq_time
-
-        # TODO: Expand acquisition times list so that each energy step has an
-        # associated time
+        self.acquisition_times = []
+        energy_steps = lo_stepping_values[
+            ["e1", "e2", "e3", "e4", "e5", "e6", "e7", "e8"]
+        ].astype(str)  # convert to string to avoid confusion with table index value
+        for step_number in range(128):
+            index = energy_steps.isin([str(step_number)]).any(axis=1).idxmax()
+            self.acquisition_times.append(lo_stepping_values.acq_time[index])
```

Comment:
can you add some comments for clarification? Appreciate it

Reply:
I agree, I'm not sure I follow this either, so it would be good to document why you're doing isin, any, and idxmax on the previous line as well.

It looks like you're doing == but using isin, are you not able to do a comparison on the arrays? `energy_steps == str(step_number)`

Reply:
This can be simplified to:

```python
index = (energy_steps == str(step_number)).any(axis=1).idxmax()
```

The `axis=1` is finding in which _row_ the step number is found, and `idxmax()` is converting that to an index number.

Reply:
As an alternative suggestion. If they are all unique, I think you might be able to use `np.nonzero(energy_steps == str(step_number))[0]` with `nonzero` giving you the indices of the entries that matched your expression.

I think the `idxmax()` is throwing me off because it seems like it is being used for more than one return value. `argwhere` or `argmax` might be suitable as well.

Reply:
Thanks. It looks like this also works:

```python
index = np.argmax(energy_steps == str(step_number), axis=1).argmax()
```

I don't really find any of these solutions to be very readable/straightforward to be honest, but perhaps this one is the easiest to follow(?)

Reply:
I still think you just want to grab the single value though and not necessarily the "max"?

https://stackoverflow.com/questions/69329249/how-do-i-find-the-index-of-an-specific-value-in-numpy-array

```python
# will return (5, 6) row/col of the first match
np.argwhere(a == 0)[0]
# I think will return 5, the row of the first match
np.argwhere(a == 0)[0][0]
```

Reply:
Right, so the `argwhere` solution would be:

```
index = np.argwhere(energy_steps == str(step_number)[0][0]
```

So of all these solutions, which is the best?

```python
index = (energy_steps == str(step_number)).any(axis=1).idxmax()
```

```python
index = np.argmax(energy_steps == str(step_number), axis=1).argmax()
```

```python
index = np.argwhere(energy_steps == str(step_number)[0][0]
```